### DISCUSSION TITLE
Will this project be viable without ephemeralForTest engine?

### DISCUSSION THREAD

**Discussion by adam-kiss-sg** [typegoose/mongodb-memory-server#706] [Q&A] [answered]
## Versions

package: mongo-memory-server 

## What is your question?

It seems like ephemeralForTest engine has been removed from mongodb 6.1.0: https://github.com/mongodb/mongo/commit/e1f267e15c92f48580bd77a6138fc4cc09be6913

I have been checking out this package, and it works great for local testing! (I still have some issues with CI due to using alpine, but that's unrelated :) )

Would be curious to know what your plans are going forward, now that the ephemeralForTest has been removed from mongodb. Do you think this project could still be viable with the wiredTiger engine?

**Comment by hasezoey** [comment, marked as answer]
> Would be curious to know what your plans are going forward, now that the ephemeralForTest has been removed from mongodb. Do you think this project could still be viable with the wiredTiger engine?

personally i still think it is "viable", as long as you dont want to deal with configuration or download yourself,
also there are still older versions supported by this package (and are currently the defaults) which have this engine (and will probably also for the next few years),

personally, `wiredTiger` is still viable locally, though a bit slower, with the right system configuration it works great,
required system configuration (at least on linux, i dont know about windows and mac) is to have `/tmp` (the default automatically created dbPath) be a `tmpfs`, for example with `systemd-tmpfiles-setup.service` or manually with `tmpfs` mount,

as for CI, some may provide services, but this package will still allow to have the same behavior across CI and locally, where you can also choose custom configuration easily,

---

as for the future, we will probably add warnings in the documentation, tsdoc itself and add runtime warnings when using a version that matches `>=6.1.0` and automatically translate to `wiredTiger`, and if 6.1.0 (or higher) will become the default, then the engine will also be changed (before it is still faster to have that engine, and only switch when required)

---

PS: i have moved this to a discussion, because i think this question is more suited for it

**Reply by adam-kiss-sg** [reply]
thanks for the detailed answer
